integration/docker/csi/alluxio/nodeserver.go

```diff
@@ -87,45 +103,203 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 	return &csi.NodePublishVolumeResponse{}, nil
 }
 
-func (ns *nodeServer) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpublishVolumeRequest) (*csi.NodeUnpublishVolumeResponse, error) {
+func nodePublishVolumeMountPod(req *csi.NodePublishVolumeRequest) (*csi.NodePublishVolumeResponse, error) {
 	targetPath := req.GetTargetPath()
+	stagingPath := req.GetStagingTargetPath()
+
+	notMnt, err := ensureMountPoint(targetPath)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	if !notMnt {
+		glog.V(4).Infoln("target path is already mounted")
+		return &csi.NodePublishVolumeResponse{}, nil
+	}
+
+	// Wait for alluxio-fuse pod finishing mount to global mount point
+	i := 0
+	for i < 10 {
+		time.Sleep(3 * time.Second)
+		command := exec.Command("bash", "-c", fmt.Sprintf("mount | grep %v | grep alluxio-fuse", stagingPath))
+		stdout, err := command.CombinedOutput()
+		if err != nil {
+			glog.V(3).Infoln("Alluxio is not mounted.")
+		}
+		if len(stdout) > 0 {
+			break
+		}
+		i++
+	}
+	if i == 10 {
+		glog.V(3).Infoln("alluxio-fuse is not mounted to global mount point in 30s.")
```

Comment:
Move this to `NodeStageVolume`?  After `NodeStageVolume` we should make sure the volume already ready in this node